### DISCUSSION TITLE
Preprints Section of Publications Page

### DISCUSSION THREAD

**Discussion by goldblum** [alshedivat/al-folio#3242] [Q&A] [answered]
I want to add a "preprints" section to the publications page.  I tried just entering "preprints" as a year, and while this does add the section, the name of the section now overlaps the title (see the picture below).  It would be great if there were an easy way to add a preprints section since so many academics have papers they want to highlight which have not been published yet.

![Screenshot from 2021-07-02 22-29-49](https://user-images.githubusercontent.com/442416/124340711-20103b00-db85-11eb-897f-7acff714025b.png)

**Comment by sdaza** [comment]
I added: 

`{% if entry.preprint %}
        <a href="{{ entry.preprint }}" class="btn btn-sm z-depth-0" role="button" target="_blank">Preprint</a>
    {% endif %}`

To this file https://github.com/sdaza/sdaza.github.com/blob/source/_layouts/bib.html

Then, the bib entries could be included in the preprint field, an example:

```
@article{voigt2020,
  title = {Trends in {{Education}}-Specific {{Differences}} in {{Disability}}-{{Free Life Expectancy}} in {{Spain}}, 2008-2017},
  author = {Voigt, Mathias and Daza, Sebastian and Ordanovich, Dariya and Palloni, Alberto},
  date = {2020},
  journaltitle = {Working Paper},
  url = {https://osf.io/preprints/socarxiv/mf6n8/},
  urldate = {2020-09-02},
  code = {https://gitlab.com/csic-echo/compression-morbidity},
  preprint = {https://osf.io/preprints/socarxiv/mf6n8/},
  status = {Under review}
}
```

You can use the same logic to add any field you want (https://sdaza.com/publications/). Using Zotero, you can add those fields in the Extra section: 

```
tex.code= https://github.com/sdaza/reentry-work-lifecourse
tex.preprint = https://osf.io/preprints/socarxiv/k5zyd/
tex.status = Under review
tex.bibtex_show = true
```

**Comment by goldblum** [comment]
@sdaza Thanks for the suggestion.  I do want a separate section for preprints though outside of other years, since those papers are often working papers and don't fall into a specific year nicely.

**Comment by sdaza** [comment]
just a short comment, it is a trend that published papers include also a preprint...

**Comment by goldblum** [comment]
Thanks.  I do think this is a solution for a different issue.

**Comment by sdaza** [comment]
yes, I agree.

**Comment by William-N-Havard** [comment, marked as answer]
You can do that quite easily by tweaking the configuration of the jekyll/scholar module

**in _config.yml add:**
sort_by: year, month
order: descending, descending
type_order: [unpublished, article, inbook, conference, incollections, inproceedings, proceedings, book, misc, techreport]
type_names: {unpublished: Pre-Prints / In Preparation}

**in _pages/publications.md file, replace everything by:**
```
<div class="publications">
 {% bibliography --template bib --group_by type,year --group_order ascending,descending %}
</div>
```

**in your bibtex**
use @unpublished instead of @article

No need to manually group by type/year when jekyll/scholar can already do it!

**Comment by goldblum** [comment]
@William-N-Havard 

Thanks so much for your help!  I tried this, but it creates a section for "Journal Articles", one for "Conference Articles", and one for "Unpublished".  It also leaves weird text near the top.  I assume I'm doing something incorrect.  Should my full _pages/publications.md file be:

```
---
layout: page
permalink: /publications/
title: publications
nav: true
---

<div class="publications">
 {% bibliography --template bib --group_by type,year --group_order ascending,descending %}
</div>
```

**Comment by sdaza** [comment]
I was able to use the configuration that @William-N-Havard suggested:

My `_config` file includes: 

``` 
sort_by: date
order: descending
type_order: [article, inbook, book, unpublished, misc, report, conference, incollections, inproceedings, proceedings]
type_names: {unpublished: working papers, report: reports, article: journal articles, 
      inproceedings: conference articles & others}
```

`publication.md` file (my bib file is `sdaza.bib`):

```
---
layout: page
permalink: /publications/
title: publications
nav: true
---

<div class="publications">
 {% bibliography -f sdaza --group_by type %}
</div>
```

I changed a bit the type headers adding these lines to _base.scss in the publication section: 

```
  h2.bibliography {
    color: $grey-color-light;
    border-top: 1px solid $grey-color-light;
    padding-top: 1rem;
    margin-top: 2rem;
    margin-bottom: 2rem;
    text-align: right;
  }
```

An example here: https://sdaza.com/publications/

**Comment by Braghadeeshln** [comment]
> You can do that quite easily by tweaking the configuration of the jekyll/scholar module
> 
> **in _config.yml add:** sort_by: year, month order: descending, descending type_order: [unpublished, article, inbook, conference, incollections, inproceedings, proceedings, book, misc, techreport] type_names: {unpublished: Pre-Prints / In Preparation}
> 
> **in _pages/publications.md file, replace everything by:**
> 
> ```
> <div class="publications">
>  {% bibliography --template bib --group_by type,year --group_order ascending,descending %}
> </div>
> ```
> 
> **in your bibtex** use @Unpublished instead of @Article
> 
> No need to manually group by type/year when jekyll/scholar can already do it!

I tried this approach, but I am unable to change Unpublished to Preprints

**Comment by riasat-sheikh** [comment]
> I tried this approach, but I am unable to change Unpublished to Preprints

In the `_config.yml` add 

```yaml
scholar:
  # other codes
  type_names: { Unpublished: Preprints }
```
